integration-tests/src/tests/client/features/access_key_nonce_for_implicit_accounts.rs

```diff
@@ -260,7 +260,22 @@ fn test_chunk_transaction_validity() {
         .persist_and_distribute_encoded_chunk(chunk, merkle_paths, receipts, validator_id)
         .unwrap();
     let res = env.clients[0].process_block_test(block.into(), Provenance::NONE);
-    assert_matches!(res.unwrap_err(), Error::InvalidTransactions);
+    match res.as_deref() {
+        Ok([h]) => {
+            let Ok(block) = env.clients[0].chain.get_block(&h) else {
+                panic!("did not find block from result!");
+            };
+            for chunk_hdr in block.chunks().iter_raw() {
+                let hash = chunk_hdr.chunk_hash();
+                let Ok(chunk) = env.clients[0].chain.mut_chain_store().get_chunk(&hash) else {
+                    continue;
+                };
+                assert_eq!(chunk.prev_outgoing_receipts().len(), 0);
+            }
```

Comment:
could you please explain this change

Reply:
So, first `process_block_test` calls in sequence `start_process_block` and then `postprocess_ready_blocks`. In `stable` the invalid transactions this test submits get failed out in `start_process_block` with `InvalidTransactions`.

`postprocess_ready_blocks` in stable eventually calls `apply_chunks_preprocessing` -> `get_update_shard_job` -> `validate_chunk_transactions`.

`validate_chunk_transactions` then was changed in nightly to return `Ok(())` with `protocol_feature_relaxed_chunk_validation` as I've pushed transaction order and similar such checks all the way down to fail in the transaction runtime.

So what happens instead is that chunks do make their way to the transaction runtime (for now they're valid) and there the transactions are ignored, resulting in chunks that contain 0 receipts.

This is what this added code is checking for -- that no receipts have been processed even if the block was successfully created/processed.